module_utils/ansible_nailgun_cement.py

```diff
@@ -476,6 +477,13 @@ def find_compute_profile(module, name, failsafe=False):
     return handle_find_response(module, compute_profile, message="No compute profile found for %s" % name, failsafe=failsafe)
 
 
+def find_compute_attribute(module, compute_profile_name, compute_resource_name, failsafe=False):
+    search = {'search': 'compute_profile="{}" and compute_resource="{}"'.format(compute_profile_name, compute_resource_name)}
+    compute_attribute = ComputeAttribute().search(set(), search)
```

Comment:
Can you change this to `ComputeAttribute(compute_profile=compute_profile, compute_resource=compute_resource).search()` Your just need to pass in the entities and not just the names.

Reply:
This would have allowed us to drop the `try except` in the module`

Reply:
I think this will add 1 request per compute_attribute to get compute_resource entity from its name before passing it to `find_compute_attribute` in the case of the compute_attribute already exist that is the nominal case if your infra don't move too much on that side and you enforce your foreman configuration on a regular basis.